libinteract/acPSN.py

```diff
@@ -0,0 +1,637 @@
+#!/usr/bin/env python
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+#
+# Copyright (C) 2020, Valentina Sora <sora.valentina1@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful, 
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Protein Structure Network calculation
+=======================================================================
+!!! Before using this module, check the integrity of the PBC
+!!! of your trajectory because it cannot handle broken PBCs.
+
+This module contains a base class to compute Protein Structure Networks
+from sets of protein structures as described in Kannan and 
+Vishveshwara, JMB, (1999)292,441-464.
+"""
+
+
+# Standard library
+import collections.abc as abc
+import copy
+import logging
+import math
+import os
+import pkg_resources
+import sys
+
+# Third-party packages
+import MDAnalysis as mda
+import MDAnalysis.analysis.distances as dist
+import networkx as nx
+import numpy as np
+
+# Set logger
+log = logging.getLogger(__name__)
+
+
+
+class AtomicContactsPSNBuilder(object):
+    """Class implementing a builder for a Protein Structure
+    Network (PSN) based on atomic contacts as described in
+    Kannan and Vishveshwara, JMB, (1999)292,441-464.
+    """
+    
+    # Selecion string for only heavy side chains atoms 
+    # (hydrogens excluded) 
+    SELECTION_STRING = "protein and not backbone and prop mass >= 3"
+
+    # Name to be given to the group of atoms used for the calculation
+    # of atomic contacts
+    SELECTION_NAME = "sidechain"
+
+
+    def __init__(self):
+        """Set up a Protein Network Analysis on a trajectory.
+        The analysis is performed either parsing the trajectory
+        frame by frame and getting the corresponding PSN(s) at
+        each step with :meth:'PSNAnalysis.iter_psns()' or
+        parsing the whole trajecory (or a subset of frames from
+        the trajectory) and storing all the resulting PSNs with 
+        :meth:'PSNAnalysis.get_psns()'.
+        """
+        pass    
+
+
+    ######################### PRIVATE METHODS #########################
+
+
+    def _set_i_min(self,
+                   i_min):
+        """Check and return a single i_min value.
+        """
+        
+        # If the i_min is neither an integer, a float or None, raise
+        # an error
+        if not isinstance(i_min, (int, float, type(None))):
```

Comment:
No, since the only reason would have been to deal with the case where the `i_mins` list was `None`, but that point is never reached anywhere in the code since in case `i_mins` is `None` a list of i_mins containing only 0.0 (i.e. `[0.0])`) is returned before the i_mins are passed to this function that checks them one by one.

I, therefore, removed `None` from the accepted types since it is a case that will never be encountered.

Reply:
I also modified the default value of the i_min to 0.0.